### ISSUE TITLE
Cannot redeclare class ...

### ISSUE THREAD

**Issue by DrSchimke** [sensiolabs-de/deprecation-detector#40]
The project to be checked has same dependencies as the deprecation-detector (e.g. nikic/php-parser), so I get following error by running the check:
PHP Fatal error:  Cannot redeclare class PhpParser\Autoloader in /local/some_project/vendor/nikic/php-parser/lib/PhpParser/Autoloader.php on line 9

In my case I just can install the compoer-vendor-stuff with --no-dev to solve the problem, but I think, this is no solution for all. Any ideas?

**Comment by slde-gorillaman** [comment]
Can you paste the full exception log? (run with -vvv option)
This way we can better determine at which point the of execution the exception occurs

**Comment by DrSchimke** [comment]
```
PHP Fatal error:  Cannot redeclare class PhpParser\Autoloader in /data/kx/vendor/nikic/php-parser/lib/PhpParser/Autoloader.php on line 9
PHP Stack trace:
PHP   1. {main}() /data/deprecation-detector/bin/deprecation-detector:0
PHP   2. Symfony\Component\Console\Application->run() /data/deprecation-detector/bin/deprecation-detector:11
PHP   3. Symfony\Component\Console\Application->doRun() /data/deprecation-detector/vendor/symfony/console/Application.php:126
PHP   4. Symfony\Component\Console\Application->doRunCommand() /data/deprecation-detector/vendor/symfony/console/Application.php:195
PHP   5. Symfony\Component\Console\Command\Command->run() /data/deprecation-detector/vendor/symfony/console/Application.php:886
PHP   6. SensioLabs\DeprecationDetector\Console\Command\CheckCommand->execute() /data/deprecation-detector/vendor/symfony/console/Command/Command.php:259
PHP   7. SensioLabs\DeprecationDetector\Detector\ViolationDetector->getViolations() /data/deprecation-detector/src/Console/Command/CheckCommand.php:144
PHP   8. SensioLabs\DeprecationDetector\Violation\ViolationChecker\ComposedViolationChecker->check() /data/deprecation-detector/src/Detector/ViolationDetector.php:46
PHP   9. array_map() /data/deprecation-detector/src/Violation/ViolationChecker/ComposedViolationChecker.php:35
PHP  10. SensioLabs\DeprecationDetector\Violation\ViolationChecker\ComposedViolationChecker->SensioLabs\DeprecationDetector\Violation\ViolationChecker\{closure}() /data/deprecation-detector/src/Violation/ViolationChecker/ComposedViolationChecker.php:35
PHP  11. SensioLabs\DeprecationDetector\Violation\ViolationChecker\MethodDefinitionViolationChecker->check() /data/deprecation-detector/src/Violation/ViolationChecker/ComposedViolationChecker.php:30
PHP  12. SensioLabs\DeprecationDetector\AncestorResolver->getClassAncestors() /data/deprecation-detector/src/Violation/ViolationChecker/MethodDefinitionViolationChecker.php:33
PHP  13. SensioLabs\DeprecationDetector\AncestorResolver->resolveInterfaceAncestors() /data/deprecation-detector/src/AncestorResolver.php:70
PHP  14. SensioLabs\DeprecationDetector\AncestorResolver->getDefinitionFile() /data/deprecation-detector/src/AncestorResolver.php:93
PHP  15. SensioLabs\DeprecationDetector\AncestorResolver->findDefinitionFileByComposer() /data/deprecation-detector/src/AncestorResolver.php:147
PHP  16. SensioLabs\DeprecationDetector\AncestorResolver->initComposerLoader() /data/deprecation-detector/src/AncestorResolver.php:172
PHP  17. include() /data/deprecation-detector/src/AncestorResolver.php:209
PHP  18. ComposerAutoloaderInit465e3d1290a32434d8df0a3014838e19::getLoader() /data/kx/vendor/autoload.php:7
PHP  19. composerRequire465e3d1290a32434d8df0a3014838e19() /data/kx/vendor/composer/autoload_real.php:49
PHP  20. require() /data/kx/vendor/composer/autoload_real.php:58
PHP  21. require() /data/kx/vendor/nikic/php-parser/lib/bootstrap.php:3
```

**Comment by slde-thespirit** [comment]
Seems to be caused by using the autoloader of the project to find definition files in AncestorResolver.

**Comment by DrSchimke** [comment]
My thoughts are: why the heck the deprecation-detector (or nikic/php-parser) includes/requires the _code to be inspected_. So the _code to be inspected_ occupies parts of the Php global context: loads conflicting classes, declares functions, ...

I think, somehow is would be better, if the code is just read (as is done by the tokenizer).

**Comment by dvitharanage** [comment]
Has this been fixed yet ? 

I've got the same issue, --no-dev does not make any difference in my case.

**Comment by smoench** [comment]
Has been fixed with #128
